extensions/notebook/src/book/bookTreeView.ts

```diff
@@ -8,45 +8,35 @@ import * as vscode from 'vscode';
 import * as path from 'path';
 import * as fs from 'fs';
 import * as yaml from 'js-yaml';
+import * as fg from 'fast-glob';
 import { BookTreeItem, BookTreeItemType } from './bookTreeItem';
 import * as nls from 'vscode-nls';
 const localize = nls.loadMessageBundle();
 
-
 export class BookTreeViewProvider implements vscode.TreeDataProvider<BookTreeItem>, azdata.nb.NavigationProvider {
 	readonly providerId: string = 'BookNavigator';
 
 	private _onDidChangeTreeData: vscode.EventEmitter<BookTreeItem | undefined> = new vscode.EventEmitter<BookTreeItem | undefined>();
 	readonly onDidChangeTreeData: vscode.Event<BookTreeItem | undefined> = this._onDidChangeTreeData.event;
-	private _tableOfContentsPath: string[];
+	private _tableOfContentPaths: string[] = [];
 	private _allNotebooks = new Map<string, BookTreeItem>();
 	private _extensionContext: vscode.ExtensionContext;
 	private _throttleTimer: any;
 	private _resource: string;
 
 	constructor(workspaceFolders: vscode.WorkspaceFolder[], extensionContext: vscode.ExtensionContext) {
 		let workspacePaths: string[] = workspaceFolders.map(a => a.uri.fsPath);
-		this._tableOfContentsPath = this.getTableOfContentFiles(workspacePaths);
-		let bookOpened: boolean = this._tableOfContentsPath && this._tableOfContentsPath.length > 0;
-		vscode.commands.executeCommand('setContext', 'bookOpened', bookOpened);
+		this.getTableOfContentFiles(workspacePaths);
 		this._extensionContext = extensionContext;
 	}
 
-	private getTableOfContentFiles(directories: string[]): string[] {
-		let tableOfContentPaths: string[] = [];
-		let paths: string[];
-		directories.forEach(dir => {
-			paths = fs.readdirSync(dir);
-			paths.forEach(filename => {
-				let fullPath = path.join(dir, filename);
-				if (fs.statSync(fullPath).isDirectory()) {
-					tableOfContentPaths = tableOfContentPaths.concat(this.getTableOfContentFiles([fullPath]));
-				} else if (filename === 'toc.yml') {
-					tableOfContentPaths.push(fullPath);
-				}
-			});
-		});
-		return tableOfContentPaths;
+	async getTableOfContentFiles(directories?: string[]): Promise<void> {
+		for (let i in directories) {
```

Comment:
so... I'd say we discourage the `i in array` syntax as it is easily confused with the `for (let value of arrays)` syntax. Can you please use `for (let dir of directories)` style syntax, or `for (let i = 0; i < directories.length; i++)` syntax instead? This is just to be consistent with our style

Reply:
Changed to for (let dir of directories) syntax :)